### PR TITLE
Nwd nnw - D4rth

### PR DESCRIPTION
**Pull request by Morfiniusz**
:)

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: nwd-nnw
🏅 Granted 6 XP for @Morfiniusz

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Morfiniusz for on-time delivery

File: homework/nwd-nnw/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,45 @@
 #pragma once
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
+    int rest;
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:4]: Better always initialize values created on stack with default value.

File: homework/nwd-nnw/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+        return lhs;
+    }
+    else if(lhs != 0 and rhs == 0){
+        return lhs;
+    } 
+    else if(lhs == 0 and rhs != 0){
+        return rhs;
+    } 
+    else{
+        return 0;
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:26]: If you return variable here, you don't need to use `else if`

File: homework/nwd-nnw/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
+    }
+    if (rhs < 0){
+        rhs *= -1;
+    }
+    if (lhs != 0 and rhs != 0){
+        return (lhs*rhs)/NWD(lhs,rhs);
+    }
+    else{
+        return 0;
+    }
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:43]: If you return variable here, you don't need to use `else if`

File: homework/nwd-nnw/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
+    if (lhs < 0){
+        lhs *= -1;
+    }
+    if (rhs < 0){
+        rhs *= -1;
+    }
+    if (lhs != 0 and rhs != 0){
+        return (lhs*rhs)/NWD(lhs,rhs);
```
Comment (nauka-programowania-MA) [inline homework/nwd-nnw/nwdNww.hpp:39]: Add space between operations like '+' '-' '!=' '==' etc...

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**